File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap2/PrintFiles.pl
#!/usr/bin/perl -w

# PrintFiles.pl

=head1

A Script for Printing Multiple PDF and/or PS Files

=description

Any of the following calls will work

    PrintFiles.pl *.pdf

    PrintFiles.pl *.PDF

    PrintFiles.pl *.ps

=cut


use strict;

die "requires at least one file name" 
    unless @ARGV;                                                   #(A)

foreach (@ARGV) {                                                   #(B)
    die "Only pdf and ps files can be printed out\n"                #(C)
        unless /\.ps|(df)$/i;                                       #(D)
    system "gs -q -dNOPAUSE ".                                      #(E)
           "-sOUTPUTFILE=- -sDEVICE=ljet4 -r600x600 ".              #(F)
           "$_".                                                    #(G)
           " -c quit | lp -d elt339";                               #(H)
}

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap3/PrintFiles.py
#!/usr/bin/env python

### PrintFiles.py

import sys                      # for argv[], exit()                #(A)
import os                       # for system()                      #(B)

if len( sys.argv ) < 2:                                             #(C)
    print "Needs at least one file name as argument"                #(D)
    sys.exit(1)                                                     #(E)

for item in sys.argv[1:]:                                           #(F)
    if item.endswith( ('.ps', '.pdf') ):                            #(G)
        cmd = "/usr/local/bin/gs -q -dPrinted -dNOPAUSE " \
              "-sOUTPUTFILE=- -sDEVICE=pswrite "          \
              "%s" % item                                           #(H)
        cmd = cmd + " -c quit | lp -d elt339"                       #(I)
        os.system( cmd )                                            #(J)
